server/fleet/agent_options.go

```diff
@@ -269,6 +276,198 @@ type osqueryOptions struct {
 	YaraDelay                           uint32 `json:"yara_delay"`
 }
 
+// NOTE: generate automatically with `go run ./tools/osquery-agent-options/main.go`
```

Comment:
did you generate this on a Linux machine? I think there's a small edge case, which I'm unsure if we want to tackle or not: osquery has certain command line flags that are os-specific (search for "control flags" [here](https://osquery.readthedocs.io/en/stable/installation/cli-flags/))

Asking because I can see the Linux-specific `hardware_disabled_types` but I can't see the macOS ones

Reply:
Oh! Good point! Yes I'm on Linux... Note that this is backend-specific, the backend PR is at #7979 . This was meant to be just the frontend PR, but it includes the backend changes as it was cut off of this branch (you may have missed my note in the PR description). I'll try to move your comment there (or I'll just copy it over).

Reply:
oops! so sorry about that